### PR TITLE
Add a check if the user is running Java 9

### PR DESCRIPTION
**Pull request by DoNotSpamPls**
I'd like to note that this has not been tested (I will test when I have some free time).
If someone can check for errors or things that don't work, I will be very grateful 😃

### CODE REVIEW

**Comment by me4502** [comment]
I don’t see what you’ve done here. It seems you’ve just duplicated the previous message, and added another method to check for it (That isn’t even called anywhere)?

Java 9 isn’t supported in Sponge as a vast majority of the Minecraft Toolchain doesn’t support it at this current point in time. Work is being done, with it depending on the replacement for LaunchWrapper in the time being.

**Review by ST-DDT — CHANGES_REQUESTED** [review]
Please follow the formatting guidelines and check that it at least compiles.

File: src/java6/java/org/spongepowered/launch/JavaVersionCheckUtils.java [hunk 1]
```diff
@@ -32,10 +32,15 @@
...
     private static final String REQUIRED_VERSION = "1.8.0_20";
 
-    private static final String ERROR_MESSAGE = "We have detected that you are JRE version %s, which is out of date!\n"
+    private static final String ERROR_MESSAGE = "We have detected that you are using JRE version %s, which is out of date!\n"
                                                 + "In order to run Sponge, you **must** be running JRE version %s (or above).\n"
                                                 + "Previous builds of JRE version 1.8 will not work with Sponge.\n"
                                                 + "This can be downloaded from Oracle: http://www.oracle.com/technetwork/java/javase/downloads/jre8-downloads-2133155.html";
+    
+    private static final String UNSUPPORTED = "We have detected that you are using JRE version %s, which is currently not supported!\n"
+                                              + "In order to run Sponge, you **must** be running JRE version %s (or above).\n"
```
Comment (Deamon5550) [inline src/java6/java/org/spongepowered/launch/JavaVersionCheckUtils.java:41]: This message does not make sense to a user running java 9.

File: src/java6/java/org/spongepowered/launch/JavaVersionCheckUtils.java [hunk 2]
```diff
@@ -51,6 +56,19 @@ public static void ensureJava8() {
...
+    
+    public static void checkForNewVersions() {
+        String version = getCurrentVersion();
+        if (getVersionValue(version) > getVersionValue(REQUIRED_VERSION)) {
+            String error = String.format(UNSUPPORTED, version, REQUIRED_VERSION);
+
+                if (!GraphicsEnvironment.isHeadless()) {
+                    JOptionPane.showMessageDialog(null, error, "PEBKACException!", JOptionPane.ERROR_MESSAGE);
+                }
+            throw new RuntimeException(error);
```
Comment (Deamon5550) [inline src/java6/java/org/spongepowered/launch/JavaVersionCheckUtils.java:68]: You've removed the if but left the print after the throw, so this doesn't compile.

File: src/java6/java/org/spongepowered/launch/JavaVersionCheckUtils.java [hunk 3]
```diff
@@ -51,6 +56,19 @@ public static void ensureJava8() {
             System.out.println("You may be running an outdated version of Java. Any crashes from Sponge may require an update to Java.");
         }
     }
+    
+    public static void checkForNewVersions() {
+        String version = getCurrentVersion();
+        if (getVersionValue(version) > getVersionValue(REQUIRED_VERSION)) {
```
Comment (Deamon5550) [inline src/java6/java/org/spongepowered/launch/JavaVersionCheckUtils.java:62]: This will cause all java 8 builds except build 20 to fail...

**Review by Deamon5550 — CHANGES_REQUESTED** [review]
Also don't forget to actually call it.

**Comment by gabizou** [comment]
Code doesn't compile, doesn't accomplish a Java 9 check, and far more to the game would fail to load, even with a sponge check.

**Pull request closed without merging**